internal/resolver/delegatingresolver/delegatingresolver_ext_test.go

```diff
@@ -0,0 +1,340 @@
+		},
+	})
+
+	wantState := resolver.State{Addresses: []resolver.Address{
+		proxyAddressWithTargetAttribute(resolvedProxyTestAddr1, targetTestAddr),
+		proxyAddressWithTargetAttribute(resolvedProxyTestAddr2, targetTestAddr),
+	}}
+
+	var gotState resolver.State
+	select {
+	case gotState = <-stateCh:
+	case <-time.After(defaultTestTimeout):
+		t.Fatal("Timeout when waiting for a state update from the delegating resolver")
+	}
+
+	if !cmp.Equal(gotState, wantState) {
+		t.Fatalf("Unexpected state from delegating resolver: %s, want %s", pretty.ToJSON(gotState), pretty.ToJSON(wantState))
+	}
+}
+
+// Tests the scenario where a proxy is configured, and the target URI scheme is
+// not "dns". The test verifies that the addresses returned by the delegating
+// resolver include the resolved proxy address and the custom resolved target
+// address as attributes of the proxy address.
+func (s) TestDelegatingResolverwithCustomResolverAndProxy(t *testing.T) {
+	hpfe := func(req *http.Request) (*url.URL, error) {
+		if req.URL.Host == targetTestAddr {
+			return &url.URL{
+				Scheme: "https",
+				Host:   envProxyAddr,
+			}, nil
+		}
+		return nil, nil
+	}
+	originalhpfe := internal.HTTPSProxyFromEnvironmentForTesting
+	internal.HTTPSProxyFromEnvironmentForTesting = hpfe
+	defer func() {
+		internal.HTTPSProxyFromEnvironmentForTesting = originalhpfe
+	}()
+
+	mrTarget := manual.NewBuilderWithScheme("test") // Manual resolver to control the target resolution.
+	target := mrTarget.Scheme() + ":///" + targetTestAddr
+	mrProxy := setupDNS(t) // Set up a manual DNS resolver to control the proxy address resolution.
+
+	tcc, stateCh, _ := createTestResolverClientConn(t)
+	if _, err := delegatingresolver.New(resolver.Target{URL: *testutils.MustParseURL(target)}, tcc, resolver.BuildOptions{}, mrTarget, false); err != nil {
+		t.Fatalf("Failed to create delegating resolver: %v", err)
+	}
+
+	mrProxy.UpdateState(resolver.State{
+		Addresses: []resolver.Address{
+			{Addr: resolvedProxyTestAddr1},
+			{Addr: resolvedProxyTestAddr2},
+		},
+		ServiceConfig: &serviceconfig.ParseResult{},
+	})
+
+	select {
+	case <-stateCh:
+		t.Fatalf("Delegating resolver invoked UpdateState before both the proxy and target resolvers had updated their states.")
+	case <-time.After(defaultTestShortTimeout):
+	}
+
+	mrTarget.UpdateState(resolver.State{
+		Addresses: []resolver.Address{
+			{Addr: resolvedTargetTestAddr1},
+			{Addr: resolvedTargetTestAddr2},
+		},
+		ServiceConfig: &serviceconfig.ParseResult{},
+	})
+
+	wantState := resolver.State{Addresses: []resolver.Address{
+		proxyAddressWithTargetAttribute(resolvedProxyTestAddr1, resolvedTargetTestAddr1),
+		proxyAddressWithTargetAttribute(resolvedProxyTestAddr1, resolvedTargetTestAddr2),
+		proxyAddressWithTargetAttribute(resolvedProxyTestAddr2, resolvedTargetTestAddr1),
+		proxyAddressWithTargetAttribute(resolvedProxyTestAddr2, resolvedTargetTestAddr2),
+	},
+		ServiceConfig: &serviceconfig.ParseResult{},
+	}
```

Comment:
Same here as the other comment about aligning matching braces.

Reply:
Done